### DISCUSSION TITLE
Neo•Geo button descriptions may be wrong

### DISCUSSION THREAD

**Discussion by ACMRGB** [darthcloud/BlueRetro#455] [Q&A]
Okay, so I just did some adjustments for a NeoBT from Humble Bazooka and I think I might have noticed an error in the way the Neo•Geo buttons are described in the WebGUI. With the way they're currently named, I get the following results:

Actual Neo•Geo button functions = The way Blue Retro describes them
A = B
B = D
C = A
D = C

I think I know what the problem is though. Starting from the bottom of the face buttons and working clockwise, the original Neo•Geo CD controller arrangement is A, C, D, B. The newer Neo•Geo pads that MagicTrashman modifies for use with original hardware have a different arrangement though, which goes B, A, C, D. So the Blue Retro interface uses the button naming scheme from the newer controllers, but gives them the function of the older controllers, if that makes sense. 

Link to an image of the original Neo•Geo CD pad:
https://images.app.goo.gl/TbSsFz3q6YZJawWa7

Link to the newer pads that MagicTrashman uses:
https://images.app.goo.gl/oLQLuKprJqJKT5iU7

Hopefully this is helpful. Had a helluva time setting up my 8BitDo controller in X-Input mode but I think I have it where I want it now.

**Comment by darthcloud** [comment]
Fixed this in the beta interface at:
https://darthcloud.github.io/BlueRetroWebCfg_Beta/advance.html
![chrome_0fRJWsDm7e](https://user-images.githubusercontent.com/3744056/205516576-e5f6543f-3807-4c1a-8d54-9c769a80d6e6.png)

BlueRetro is base on the layout of the original NeoGeo CD pad

Thanks for reporting!! and sorry for delay.

This will merge into the regular config site a bit later.

**Reply by darthcloud** [reply]
now fixed in blueretro.io aswell

**Reply by ACMRGB** [reply]
Thanks for fixing this! Really appreciate the work you put in to BlueRetro. :)
